### PR TITLE
feat/Examples about priority-based semi-deterministic leader election.

### LINKED ISSUE #389: Examples about priority-based semi-deterministic leader election for users.
### Your question

SOFAJRaft has already supportted priority-based semi-deterministic leader election in the version of 1.3.0.But it didn't updated its example codes for users.

### Your scenes

So, I think we can provide some examples for priority-based semi-deterministic leader election for users.

### Your advice

In the example module, we can provide some examples about priority-based semi-deterministic leader election.

### Environment

- SOFAJRaft version:
- JVM version (e.g. `java -version`):
- OS version (e.g. `uname -a`):
- Maven version:
- IDE version:

### PR DESCRIPTION
**Pull request by zongtanghu**
### Motivation:

I think it's much more helpful to users if there are priority-based semi-deterministic leader election examples.

### Modification:

I Add a detailed example in the jraft-exmaple module.

Describe the idea and modifications you've done.

### Result:

Fixes #389 .

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Comment by zongtanghu** [comment]
I'm very sorry this feature's pr is too later. @killme2008 @fengjiachun

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @author zongtanghu
+ */
+public class PriorityElectionBootstrap {
+
+    // Start elections by 3 instance. Note that if multiple instances are started on the same machine,
+    // the first parameter `dataPath` should not be the same.
+    public static void main(final String[] args) {
+        if (args.length < 4) {
+            System.out
+                .println("Useage : java com.alipay.sofa.jraft.example.priorityelection.ElectionBootstrap {dataPath} {groupId} {serverId} {initConf}");
```
Comment (fengjiachun) [inline jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:30]: PriorityElectionBootstrap ?

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            System.out
+                .println("Example: java com.alipay.sofa.jraft.example.priorityelection.ElectionBootstrap /tmp/server1 election_test 127.0.0.1:8081::100 127.0.0.1:8081::100,127.0.0.1:8082::40,127.0.0.1:8083::40");
```
Comment (fengjiachun) [inline jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:32]: PriorityElectionBootstrap?

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                System.out.println("[PriorityElectionBootstrap] Leader start on term: " + leaderTerm);
+            }
+
+            @Override
+            public void onLeaderStop(long leaderTerm) {
+                System.out.println("[PriorityElectionBootstrap] Leader stop on term: " + leaderTerm);
+            }
+        });
+        node.init(priorityElectionOpts);
+    }
```
Comment (fengjiachun) [inline jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:60]: 这个 demo 如何看出优先级高的被选成 leader？

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java (hunk 1 above)
Comment (zongtanghu) [reply jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:30]: Okay, I will fix it!

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java (hunk 2 above)
Comment (zongtanghu) [reply jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:32]: Okay, I will fix it!

File: jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java (hunk 3 above)
Comment (zongtanghu) [reply jraft-example/src/main/java/com/alipay/sofa/jraft/example/priorityelection/PriorityElectionBootstrap.java:60]: Okay, I will print the priority value, ip and port information in this node in next commits.

**Comment by zongtanghu** [comment]
I fix some small issues in this pr, please help to review again.Thanks. @fengjiachun

**Pull request merged**